File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    let number = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    
    let temp = "";
    let answer = "";
    
    for (let ch of s) {
        if (ch >= "0" && ch <= "9") {
            answer += ch;
        } else {
            temp += ch;
            switch(temp) {
                case "zero":
                    answer += 0;
                    temp = "";
                    break;
                case "one":
                    answer += 1;
                    temp = "";
                    break;
                case "two":
                    answer += 2;
                    temp = "";
                    break;
                case "three":
                    answer += 3;
                    temp = "";
                    break;
                case "four":
                    answer += 4;
                    temp = "";
                    break;
                case "five":
                    answer += 5;
                    temp = "";
                    break;
                case "six":
                    answer += 6;
                    temp = "";
                    break;
                case "seven":
                    answer += 7;
                    temp = "";
                    break;
                case "eight":
                    answer += 8;
                    temp = "";
                    break;
                case "nine":
                    answer += 9;
                    temp = "";
                    break;
            }
        }
    }
    
    return Number(answer);
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    words = ["zero", "one", "two", "three","four","five","six","seven", "eight", "nine"]
    for i, word in enumerate(words):
        s = s.replace(word, str(i))
    return int(s)
     


# 1. word 배열에 각 영단어 담기
# 2. for문을 돌면서 enumerate를 사용해서 리스트에서의 인덱스, 값을 같이 꺼냄
# 3. word를 마주치면 인덱스 값인 숫자(str)로 치환
# 4. 현재 str 이니까 int로 변환